src/pages/signup.js

```diff
@@ -27,207 +27,199 @@ const SignUp = () => {
-  const [password, setPassword] = useState("")
-  const [displayName, setDisplayName] = useState("")
-  const [error, setError] = useState(null)
-  const [passwordStrong, setPasswordStrong] = useState(false)
-  const [isLoading, setIsLoading] = useState(false)
+  const [form, setForm] = useState({email: "",
+    password: "",
+    message: { text: "", type: "" },
+    passwordStrong: false,
+    isLoading: true,
+    onVerifyLink: false})
 
-  // checks that all user inputs are not empty
-  const validateInputs = () => {
-    if (!passwordStrong) {
-      setError("Please select a more complex password")
-      return false
-    }
-
-    if (email === "" || password === "" || displayName === "") {
-      setError("Error signing up with email and password")
-      return false
-    }
-
-    return true
-  }
-
-  // generate a new document for a new user
-  const createUserWithEmailAndPasswordHandler = async (event, email, password) => {
-    // if the user has attempted every input
-
-    setIsLoading(true)
+  // Generate a new document for a new user
+  const sendUserVerificationEmail = (event) => {
+    setForm({ ...form, isLoading: true })
     event.preventDefault()
     try {
-      const { user } = await auth.createUserWithEmailAndPassword(email, password)
-
-      // manage the data to save to the documents
-      const photoURL =
-        "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAARMAAAC3CAMAAAAGjUrGAAAAA1BMVEX///+nxBvIAAAAR0lEQVR4nO3BAQ0AAADCoPdPbQ8HFAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPBgxUwAAU+n3sIAAAAASUVORK5CYII="
-      const splitNames = displayName.split(" ")
-      const firstName = splitNames[0]
-      const lastName = String(splitNames.slice(1, splitNames.length)).replace(/,/g, " ")
-
-      // to add more fields just input them into the genUserDoc object parameter.
-      generateUserDocument(user, {
-        displayName,
-        photoURL,
-        firstName,
-        lastName,
-      })
-
-      // update the user that will be stored in state
-      user
-        .updateProfile({
-          displayName,
-          photoURL,
-        })
-        .then((res) => {
-          dispatch({ type: "LOGIN", user })
-
-          navigate("/")
-        })
+      // Send the verification email and tell the user to check their inbox
+      verifyEmail(form.email, "signup")
+      setForm({ ...form, message: { text: "Please check your email to complete your signup", type: "success" } })
     } catch (error) {
-      setIsLoading(false)
+      setForm({ ...form, isLoading: false })
 
       // handle and display the various errors to the user
       if (error.code === "auth/account-exists-with-different-credential") {
-        setError("An account already exists under that email address")
-        console.error("An account already exists under that email address", error)
+        setForm({ ...form, message: { text: "An account already exists under that email address", type: "error" } })
```

Comment:
Is there an error for if you try to sign up with an email that has not been verified yet?